lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -2306,86 +2306,91 @@ bool SwiftLanguageRuntime::GetDynamicTypeAndAddress_Existential(
     return false;
   }
 
+  // This scope is needed because the validation code will call PushLocalBuffer,
+  // so we need to pop it before that call.
+  {
+    if (use_local_buffer)
+      PushLocalBuffer(
+          existential_address,
+          llvm::expectedToOptional(in_value.GetByteSize()).value_or(0));
+    auto defer = llvm::make_scope_exit([&] {
+      if (use_local_buffer)
+        PopLocalBuffer();
```

Comment:
FWIW, would it make sense to create an RAII data structure here so you don't need to manually do the whole scope_exit thing?